KULRICE-14084 Changes to use serverPaging

sonamuthu opened the pull request:
Changes to use serverSidePaging

kualiCoeusCi commented:
Can one of the admins verify this patch?

leoherbie commented on coeus-impl/src/main/java/org/kuali/coeus/org/kuali/rice/krad/uif/container/WizardGroup.java:
```diff
@@ -15,6 +15,7 @@
  */
 package org.kuali.coeus.org.kuali.rice.krad.uif.container;
 
+import org.kuali.kra.external.HashMapElement;
```
remove this import please

sonamuthu replied:
Will remove it. There are a couple of more unused imports that I'll cleanup

leoherbie commented on coeus-impl/src/main/java/org/kuali/coeus/propdev/impl/person/ProposalDevelopmentPersonnelController.java:
```diff
@@ -280,6 +281,12 @@ public ModelAndView certifyAnswers(@ModelAttribute("KualiForm") ProposalDevelopm
     }
 
+    @Transactional @RequestMapping(value = "/proposalDevelopment", params = "methodToCall=retrieveCollectionPage")
+    public ModelAndView retrieveCollectionPage(@ModelAttribute("KualiForm") DocumentFormBase form, BindingResult result, HttpServletRequest request,
```
remove result, request, response as parameters since they are unused.

sonamuthu replied:
Will remove these

blackcathacker replied:
So what about this method. I know the wizard is used in budget as well which is a different mapping. Do we need this there as well?

sonamuthu replied:
This one too is not required. I did not notice that there is already a retrieveCollectionPage in the ProposalDevelopmentCoreController and ProposalBudgetCommonController which are being used. So this was duplicate code. Thanks for noticing.

blackcathacker commented on coeus-impl/src/main/resources/org/kuali/coeus/propdev/impl/person/ProposalPersonnelPage.xml:
```diff
@@ -54,7 +54,7 @@
 
     <bean id="PropDev-PersonnelPage-Wizard" parent="Kc-Wizard"
-          p:fieldBindingObjectPath="addKeyPersonHelper">
+          p:fieldBindingObjectPath="addKeyPersonHelper" p:retrieveViaAjax="true" >
```
Is this required for the wizard to work? The wizard is used in a few different places in KC and the changes seem to be both shared and specific to the key personnel usage. Will the other usages break without similar changes?

sonamuthu replied:
retrieveViaAjax is not required here for the wizard to work with key personnel. Removing this from here. Tested it using the Proposal Key Personnel Page.

blackcathacker commented:
test this please

blackcathacker commented:
test this please

blackcathacker commented on coeus-impl/src/main/java/org/kuali/coeus/propdev/impl/person/ProposalDevelopmentPersonnelController.java:
```diff
@@ -33,6 +33,7 @@
 import org.kuali.rice.krad.uif.util.ObjectPropertyUtils;
 import org.kuali.rice.krad.web.controller.MethodAccessible;
 import org.kuali.rice.krad.web.form.DocumentFormBase;
+import org.kuali.rice.krad.web.form.UifFormBase;
```
Remove the unused import here.

blackcathacker commented on coeus-impl/src/main/resources/org/kuali/coeus/common/wizard/impl/WizardCommon.xml:
```diff
@@ -134,11 +134,17 @@
-            <map>
+        <property name="useServerPaging" value="true"/>
+        <property name="layoutManager.richTable.render" value="false"/>
+        <property name="layoutManager.richTable.templateOptions">
```
We much prefer p: to properties.

blackcathacker commented:
Needs to be rebased to be merged.

gathreya commented:
looks like this needs to be rebased before we can merge.

sonamuthu commented:
Please do not rebase or merge this pull request yet. Since I created the pull request, I have found more issues with server side paging which I am working through. 

Would you recommend closing this pull request and I will create a new one after all issues are resolved?

leoherbie commented:
No lets keep it open and we'll just continue in the PR.

sonamuthu commented:
Part of this fix is in Rice which is awaiting review (KULRICE-14084). Return To Proposal and Editing/Viewing of proposal will work after that change is pulled in.

gathreya commented:
where is this at? When can we get this into KC?

sonamuthu commented:
I am not sure what you mean by where is this at? I had updated the KC code 9 days ago and rebased the pull request. The Rice change went in the next day revision 49509.

gathreya commented:
We did not hear anything back regarding the status of 14084 since you said it was in review so we were unsure what to do with this PR. @bsmith83 should we merge this PR then?

bsmith83 commented:
This should not be merged without the rice update or without that as override in KC

gathreya commented:
Shall I close this PR ?

blackcathacker commented:
I think so. Label it for possible future review?

blackcathacker commented:
Closing for now.
